src/midiEditorCommands.cpp

```diff
@@ -134,35 +146,46 @@ struct MidiNote {
 	static const MidiNote get(MediaItem_Take* take, int index, ReqParams params) {
 		MidiNote note;
 		double start, end;
-		MIDI_GetNote(take, index,
+		if (MIDI_GetNote(take, index,
 			params.selected ? &note.selected : nullptr,
 			params.muted ? &note.muted : nullptr,
 			params.start ? &start: nullptr,
 			params.end ? &end: nullptr,
 			params.channel ? &note.channel: nullptr,
 			params.pitch ? &note.pitch: nullptr,
 			params.velocity ? &note.velocity : nullptr
-		);
-		if (params.start) {
-			start = MIDI_GetProjTimeFromPPQPos(take, start);
-			note.start = start;
-		}
-		if (params.end) {
-			end = MIDI_GetProjTimeFromPPQPos(take, end);
-			note.end = end;
+		)) {
+			if (params.start) {
+				start = MIDI_GetProjTimeFromPPQPos(take, start);
+				note.start = start;
+			}
+			if (params.end) {
+				end = MIDI_GetProjTimeFromPPQPos(take, end);
+				note.end = end;
+			}
+			note.index = index;
+		} else {
+			note.start = note.end = DBL_MAX;
 		}
-		note.index = index;
 		return note;
 	}
+
+	static const int getCount(MediaItem_Take* take) {
+		int count = 0;
+		while (MIDI_GetNote(take, count, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr, nullptr)) {
+			++count;
+		}
+		return count;
+	}
 };
 
 struct MidiEventListData { 
-	int index;
-	double position;
+	int index = -1;
+	double position = -1.0;
 	string message;
-	int offVel;
-	double length;
-	bool selected;
+	int offVel = -1;
+	double length = -1.0;
+	bool selected = -1;
```

Comment:
We shouldn't be putting -1 in a bool even if C++ allows it. We could just let this one be false by default - there are plenty of other ways to tell it is invalid - or we could use int. I'd prefer the former personally.